extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -48,9 +49,26 @@ export class NotebookService implements INotebookService {
 	 * Launch notebook with file path
 	 * @param notebook the path of the notebook
 	 */
-	launchNotebook(notebook: string | NotebookInfo): Thenable<azdata.nb.NotebookEditor> {
-		return this.getNotebookFullPath(notebook).then(notebookPath => {
-			return this.showNotebookAsUntitled(notebookPath);
+	async launchNotebook(notebook: string | NotebookInfo): Promise<azdata.nb.NotebookEditor> {
+		const notebookPath = await this.getNotebookFullPath(notebook);
+		return await this.showNotebookAsUntitled(notebookPath);
+	}
+
+	/**
+	 * Inserts cell code given by {@param cellStatements} in an existing notebook given by {@param notebook} file path at the location
+	 * {@param insertionPosition} and then launches the edited notebook.
+	 *
+	 * @param notebook - the path to notebook that needs to be launched
+	 * @param cellStatements - array of statements to be inserted in a cell
+	 * @param insertionPosition - the position at which cells are inserted. Default is a new cell at the beginning of the notebook.
+	 */
+	async launchNotebookWithEdits(notebook: string, cellStatements: string[], insertionPosition: number = 0): Promise<void> {
```

Comment:
This is called without being awaited in a sync method.

Reply:
fixed